app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/nsd/NsdServicesAdapter.kt

```diff
@@ -0,0 +1,28 @@
+package co.netguru.baby.monitor.client.feature.communication.nsd
+
+import android.net.nsd.NsdServiceInfo
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import co.netguru.baby.monitor.client.R
+
+class NsdServicesAdapter(private val onServiceClick: (nsdServiceInfo: NsdServiceInfo) -> Unit) :
+    ListAdapter<NsdServiceInfo, ServiceViewHolder>(NsdServiceDiffUtil()) {
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ServiceViewHolder {
+        val view = View.inflate(parent.context, R.layout.found_service_item, null)
+        val layoutParams = RecyclerView.LayoutParams(
+            ViewGroup.LayoutParams.MATCH_PARENT,
+            view.resources.getDimension(R.dimen.nsd_service_item_height).toInt()
+        )
+        view.layoutParams = layoutParams
+        return ServiceViewHolder(view)
+    }
+
+    override fun onBindViewHolder(holder: ServiceViewHolder, position: Int) {
+        holder.bind(getItem(position))
+        holder.itemView.setOnClickListener {
```

Comment:
Better approach is to add onClickListeners in `onCreateViewHolder`. It'll not be attached to every single item in recycler.